### PR TITLE
Enhance subscript expression

### PR DESCRIPTION
**Pull request by jievince**
Fix https://github.com/vesoft-inc/nebula-graph/issues/694

### CODE REVIEW

**Review by yixinglu — APPROVED** [review]

File: src/common/expression/SubscriptExpression.cpp [hunk 1]
```diff
@@ -56,6 +56,45 @@ const Value& SubscriptExpression::eval(ExpressionContext &ctx) {
...
+                }
+            }
+            return Value::kNullValue;
+        }
+        if (lvalue.isEdge()) {
+            if (!rvalue.isStr()) {
+                break;
+            }
+            DCHECK(!rvalue.getStr().empty());
+            if (rvalue.getStr()[0] == '_') {
```
Comment (Shylock-Hg) [inline src/common/expression/SubscriptExpression.cpp:80]: The compare is extra.

File: src/common/expression/SubscriptExpression.cpp [hunk 2]
```diff
@@ -56,6 +56,45 @@ const Value& SubscriptExpression::eval(ExpressionContext &ctx) {
...
         }
+        if (lvalue.isVertex()) {
+            if (!rvalue.isStr()) {
+                break;
+            }
+            if (rvalue.getStr() == kVid) {
+                result_ = lvalue.getVertex().vid;
+                return result_;
+            }
+            for (auto &tag : lvalue.getVertex().tags) {
```
Comment (Shylock-Hg) [inline src/common/expression/SubscriptExpression.cpp:67]: Could try `Vertex::getProp`.

File: src/common/expression/SubscriptExpression.cpp [hunk 3]
```diff
@@ -56,6 +56,45 @@ const Value& SubscriptExpression::eval(ExpressionContext &ctx) {
...
+                } else if (rvalue.getStr() == kDst) {
+                    result_ = lvalue.getEdge().dst;
+                } else if (rvalue.getStr() == kRank) {
+                    result_ = lvalue.getEdge().ranking;
+                } else if (rvalue.getStr() == kType) {
+                    result_ = lvalue.getEdge().name;
+                }
+                return result_;
+            }
+            auto iter = lvalue.getEdge().props.find(rvalue.getStr());
```
Comment (Shylock-Hg) [inline src/common/expression/SubscriptExpression.cpp:92]: ditto.

File: src/common/expression/SubscriptExpression.cpp (hunk 2 above)
Comment (jievince) [reply src/common/expression/SubscriptExpression.cpp:67]: I didn't find method called Vertex::getProp. Where is it?

Comment (Shylock-Hg) [reply src/common/expression/SubscriptExpression.cpp:67]: Could had one.

**Review by czpmango — APPROVED** [review]

**Pull request merged**